_docs/policies/tests.md

```diff
@@ -0,0 +1,47 @@
+# `status-go` Test Policy
+
+- [Creating Tests](#creating-tests)
+- [Flaky Tests](#flaky-tests)
+
+## Creating Tests
+
+- All new functionality MUST be introduced with tests that:
+  - Prove that the functionality performs as described
+  - Can be falsified
+  - Are resistant to fuzzing
+- All new tests MUST BE validated via 1000 local tests. Ensuring that the test runs and passes consistently every time gives confidence that the test is not flaky.
+    - add the `-count` flag to the test command eg: `-count 1000`
+    - OR wrap your test in a for loop
+        ```go
+        func TestTheThing(t *testing.T) {
+            for i := 1; i < 1000; i++ {
+                fmt.Println("Test Run", i)
+                // your test body goes in here
+            }
+        }
+        ```
+## Flaky Tests
+
+- All flaky tests / failing tests must be resolved
+  - Do not introduce flaky tests to the codebase
+- Steps to resolving flaky test
+  - Ensure that the test is definitely flaky and that it wasn’t you that introduced a bug that made the test flaky.
+      - Is a new test you’ve written flaky?
+        - Ok, you need to fix that before merge is acceptable.
+      - Has an old test become flaky?
+        - Ok, did you touch this test or the functionality it tests? If yes, you need to fix that before merge is acceptable. Sorry.
```

Comment:
If an old test becomes flaky, it will most likely make it through the CI unnoticed due to the retry mechanism. What we should do, is:

- Before merging, check rerun reports. If there are any, then, proceed as follows:
  - If the test does not appear in the flaky tests github issues or in the last three nightly test runs,
    - it is most likely that the flakiness was introduced by your changes and needs to be addressed before proceeding with the merge.
  - If the test is already documented as a flaky test (appears in the github issues or in the nightly test runs),
    - add a comment to the existing issue, or open a new issue if one does not already exist.